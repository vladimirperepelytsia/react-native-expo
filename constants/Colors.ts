export const Colors = {
  light: {
    text: '#11181C',
    background: '#fff',
    baseBorderColor: '#e3e3e3',
    green: '#40BF80',
    tint: '#222',
    icon: '#687076',
    tabIconDefault: '#687076',
    tabIconSelected: '#222',
    Switch: {
      trackColor: { true: '#006ae6', false: '#e3e3e3' },
      thumbColor: '#fff',
    },
  },
  dark: {
    text: '#ECEDEE',
    background: '#151718',
    baseBorderColor: '#e3e3e3',
    green: '#40BF80',
    tint: '#fff',
    icon: '#9BA1A6',
    tabIconDefault: '#9BA1A6',
    tabIconSelected: '#fff',
    Switch: {
      trackColor: { true: '#006ae6', false: '#fff' },
      thumbColor: '#fff',
    },
  },
};

export default Colors;
