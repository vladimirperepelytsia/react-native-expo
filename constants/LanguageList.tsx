const LanguageList = [
  {
    id: 1,
    title: 'English',
    value: 'en',
  },
  {
    id: 2,
    title: 'Українська',
    value: 'ua',
  },
];

export default LanguageList;
