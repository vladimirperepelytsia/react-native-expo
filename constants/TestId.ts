export const testIdHomePage = 'homePage';
export const testIdSettingPage = 'SettingPage';
export const testIdButtonBack = 'ButtonBack';
