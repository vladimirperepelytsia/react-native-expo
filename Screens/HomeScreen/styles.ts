import { StyleSheet } from 'react-native';

import type { ColorSchemeName } from 'react-native/Libraries/Utilities/Appearance';

import Colors from '@/constants/Colors';

const getStyles = ({ theme }:{ theme?: ColorSchemeName }) => StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors[theme ?? 'light'].background,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default getStyles;
