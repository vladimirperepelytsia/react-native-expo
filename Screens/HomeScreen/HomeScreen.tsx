import { View } from 'react-native';
import { useTranslation } from 'react-i18next';

import getStyles from './styles';

import AppText from '@/components/AppText/AppText';
import { testIdHomePage } from '@/constants/TestId';
import { useAppSelector } from '@/store/hooks/useApp';
import selectCurrentTheme from '@/store/slices/theme/selectors';

export default function HomeScreen() {
  const { t } = useTranslation();
  const theme = useAppSelector(selectCurrentTheme);
  const styles = getStyles({ theme });

  return (
    <View style={styles.container} testID={testIdHomePage}>
      <AppText>
        {`${t('home.pageTitle')}`}
      </AppText>
    </View>
  );
}
