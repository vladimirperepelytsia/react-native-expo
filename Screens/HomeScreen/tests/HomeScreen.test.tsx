import HomeScreen from '../HomeScreen';

import { render } from '@/__tests__/test-utils';
import { testIdHomePage } from '@/constants/TestId';

describe('<HomeScreen />', () => {
  test('should render component HomeScreen', () => {
    const tree = render(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should render component HomeScreen with text', () => {
    const { getByTestId } = render(<HomeScreen />);
    getByTestId(testIdHomePage);
  });
});
