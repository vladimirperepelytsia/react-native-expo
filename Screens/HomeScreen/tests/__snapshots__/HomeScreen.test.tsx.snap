// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> should render component HomeScreen 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="homePage"
>
  <Text
    style={
      [
        {
          "color": "#11181C",
          "fontFamily": "UbuntuRegular",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    Home page
  </Text>
</View>
`;
